logging/lazy_fields.go

```diff
@@ -0,0 +1,189 @@
+package logging
+
+import (
+	"encoding/hex"
+	"fmt"
+	spec "github.com/attestantio/go-eth2-client/spec/phase0"
+	specqbft "github.com/bloxapp/ssv-spec/qbft"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+	protocolp2p "github.com/bloxapp/ssv/protocol/v2/p2p"
+	"github.com/dgraph-io/ristretto"
+	"math/big"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/bloxapp/ssv/network/records"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	messageNotLogged = "heavy field not logged"
+	showHeavyFields  = false
+)
+
+func init() {
+	if os.Getenv("SHOW_HEAVY_FIELDS") == "true" {
+		showHeavyFields = true
+	}
+
+	//other checks, f.e. command line args or config file
+}
+
+func LazyBigInt(key string, val *big.Int) zapcore.Field {
+	if showHeavyFields {
+		return zap.String(key, val.String())
+	}
+
+	return zap.String(key, messageNotLogged)
+}
+
+func LazyHash(key string, val common.Hash) zapcore.Field {
+	if showHeavyFields {
+		return zap.String(key, val.Hex())
+	}
+
+	return zap.String(key, messageNotLogged)
+}
+
+func LazyBytesHash(key string, val []byte) zapcore.Field {
+	if showHeavyFields {
+		return zap.String(key, hex.EncodeToString(val))
+	}
+
+	return zap.String(key, messageNotLogged)
+}
+
+func LazyUrl(key string, val url.URL) zapcore.Field {
```

Comment:
I think many of these can be replaced by something like 
```suggestion
func LazyStringer(key string, str fmt.Stringer) zapcore.Field {
```